### ISSUE TITLE
Eleventy 1.0.1-canary.3 tries to process my config files even though I've told it not to

### ISSUE THREAD

**Issue by MWDelaney** [11ty/eleventy#2267]
**Describe the bug**
I believe this is due to the intricacies of my setup, but here we go anyway:
On [the ZeroPoint dev branch](https://github.com/MWDelaney/ZeroPoint/tree/dev) I've configured [two custom template languages](https://github.com/MWDelaney/ZeroPoint/blob/dev/src/config/templateLanguages.js), `scss` and `js`.

Upgrading from 1.0 to 1.0.1-canary.3 causes the previously working template language configs to fail with errors similar to the following:

```
[11ty] 1. Having trouble rendering js template ./src/config/collections.js (via TemplateContentRenderError)
[11ty] 2. Having trouble compiling template ./src/config/collections.js (via TemplateContentCompileError)
[11ty] 3. Cannot use 'in' operator to search for 'then' in undefined (via TypeError)
```

This appears to be because Eleventy is ignoring the [json file I put in the config directory](https://github.com/MWDelaney/ZeroPoint/blob/dev/src/config/config.json) which had been working to exclude `config` in 1.0.

This is a difficult situation to describe helpfully so please let me know what I can do to better illuminate the issue.

**To Reproduce**

1. Clone [ZeroPoint](https://github.com/MWDelaney/ZeroPoint) and checkout the dev branch
2. Upgrade Eleventy to 1.0.1-canary.3 in package.json and run `npm install` to install the updated dependencies
3. Run `npm run dev`

**Expected behavior**
Eleventy 1.0.1 should ignore the javascript files in `config` like 1.0 did

**Environment:**

- OS and Version: Mac
- Eleventy Version 1.0 and 1.0.1-canary.3

Thanks!

**Comment by pdehaan** [comment]
Weird.
I was able to reproduce in 1.0.1-**canary.3** and 2.0.0-**canary.2**, but not in 1.0.0.
But if I remember correctly, I was seeing the error in my log was "./src/config/filters.js"

**Comment by schneyra** [comment]
I have a similar error in 2.0.0-canary.4 with my custom extension handler for .scss-files. It tries to compile files that should not be compiled.

**Comment by zachleat** [comment]
There are a few different ways to opt out processing a template. Note that `permalink: false` does not opt out of processing (only prevents writing a template to disk) and is superfluous if you’re already ignoring a template.

Typically you’d added a glob to via standard ignore conventions (`.eleventyignore` or `eleventyConfig.ignores`): https://www.11ty.dev/docs/ignores/

But, in this specific case, you can return a falsy value from the `compile` function in your custom template language options. AND we had a regression in the fix for #2217 for this specific issue, so I appreciate you filing it! Definitely want to get this patched for 1.0.1.

Related: https://www.11ty.dev/docs/languages/custom/#compile

**Comment by zachleat** [comment]
This will ship with 1.0.1 and 2.0.0-canary.5
